code/clue.js

```diff
@@ -7,53 +7,332 @@ const mrGreen = {
+  let weaponsImage = document.getElementById('weaponsImage');
+  weaponsImage.src = mystery.weapons.image;
+  document.getElementById('weaponMaterial').innerHTML = `Made of ${mystery.weapons.material}`
+};
+
+const loadingpickWeapon = () => {
+
+  //Decides animation timer
+  const loaderTime = setTimeout(pickWeapon, 1500);
+  //Hides the cards first side 
+  document.getElementById('start-display-card-weapon').style.display = "none";
+  //Hides the card information when first when clicked (because the animation will be shown first)
+  document.getElementById('weapon-card-wrapper').style.display = "none";
+  //Displays the animation
+  document.getElementById('loader-wrapper-weapon').style.display = "flex";
+  document.getElementById('weaponCard').style.background = '#900d0d';
+}
+// Telling the function to start when card onclicks
+document.getElementById("weaponCard").onclick = loadingpickWeapon;
+
+
+// CREATE FUNCTIONS pickRoom
+const pickRoom = () => {
+  // Tells the animation to not be shown until card is clicked
+  document.getElementById('loader-wrapper-room').style.display = "none";
+  document.getElementById('room-card-wrapper').style.display = "block";
+
+  // This will randomly select a room from the rooms. And add that to the mystery object.
+  mystery.rooms = randomSelector(rooms);
+
+  //This will change styling 
+  document.getElementById('roomCard').style.background = '#e7cb96';
+  document.getElementById('roomName').style.padding = '15px';
+  document.getElementById('roomName').innerHTML = `In the ${mystery.rooms}`;
+  let roomName = document.getElementById('roomName');
+};
+console.log(pickRoom);
+
+const loadingpickRoom = () => {
+  //Decides animation timer
+  const loaderTime = setTimeout(pickRoom, 1500);
+  //Hides the cards first side 
+  document.getElementById('start-display-card-room').style.display = "none";
+  //Hides the card information when first when clicked (because the animation will be shown first)
+  document.getElementById('room-card-wrapper').style.display = "none";
+  //Displays the animation
+  document.getElementById('loader-wrapper-room').style.display = "flex";
+  document.getElementById('roomCard').style.background = "#900d0d";
+}
+// Telling the function to start when card onclicks
+document.getElementById("roomCard").onclick = loadingpickRoom;
+
+// COMMENTED OUT ALL OF THOSE TO BE ABLE TO RUN ANIMATION - To show all of this cool stuff in the browser
+//document.getElementById('killerCard').onclick = pickKiller;
+//document.getElementById('weaponCard').onclick = pickWeapon;
+//document.getElementById('roomCard').onclick = pickRoom;
 
 // STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+// Get open modal button
+const mysteryImage = document.getElementById('mystery.Image');
+
+const revealMystery = () => {
```

Comment:
Tip 💡 Instead of getting an undefined message if the user clicks on "Reveal the mystery" a little to fast: an easy way to validate if all cards are choosen is for example by doing an if-else statement in this function 😄